src/factory/TaikoDaoFactory.sol

```diff
@@ -0,0 +1,330 @@
+        OptimisticTokenVotingPlugin optimisticTokenVotingPlugin;
+        // Plugin repo's
+        PluginRepo multisigPluginRepo;
+        PluginRepo emergencyMultisigPluginRepo;
+        PluginRepo optimisticTokenVotingPluginRepo;
+        // Other
+        PublicKeyRegistry publicKeyRegistry;
+        DelegationWall delegationWall;
+    }
+
+    /// @notice Thrown when attempting to call deployOnce() when the DAO is already deployed.
+    error AlreadyDeployed();
+
+    DeploymentSettings settings;
+    Deployment deployment;
+
+    /// @notice Initializes the factory and performs the full deployment. Values become read-only after that.
+    /// @param _settings The settings of the one-time deployment.
+    constructor(DeploymentSettings memory _settings) {
+        settings = _settings;
+    }
+
+    function deployOnce() public {
+        if (address(deployment.dao) != address(0)) revert AlreadyDeployed();
+
+        IPluginSetup.PreparedSetupData memory preparedMultisigSetupData;
+        IPluginSetup.PreparedSetupData memory preparedEmergencyMultisigSetupData;
+        IPluginSetup.PreparedSetupData memory preparedOptimisticSetupData;
+
+        // DEPLOY THE DAO (The factory is the interim owner)
+        DAO dao = prepareDao();
+        deployment.dao = dao;
+
+        // DEPLOY THE PLUGINS
+        (deployment.multisigPlugin, deployment.multisigPluginRepo, preparedMultisigSetupData) = prepareMultisig(dao);
+
+        (deployment.emergencyMultisigPlugin, deployment.emergencyMultisigPluginRepo, preparedEmergencyMultisigSetupData)
+        = prepareEmergencyMultisig(dao, deployment.multisigPlugin);
+
+        (
+            deployment.optimisticTokenVotingPlugin,
+            deployment.optimisticTokenVotingPluginRepo,
+            preparedOptimisticSetupData
+        ) = prepareOptimisticTokenVoting(
+            dao, address(deployment.multisigPlugin), address(deployment.emergencyMultisigPlugin)
+        );
+
+        // APPLY THE INSTALLATIONS
+        grantApplyInstallationPermissions(dao);
+
+        applyPluginInstallation(
+            dao, address(deployment.multisigPlugin), deployment.multisigPluginRepo, preparedMultisigSetupData
+        );
+        applyPluginInstallation(
+            dao,
+            address(deployment.emergencyMultisigPlugin),
+            deployment.emergencyMultisigPluginRepo,
+            preparedEmergencyMultisigSetupData
+        );
+        applyPluginInstallation(
+            dao,
+            address(deployment.optimisticTokenVotingPlugin),
+            deployment.optimisticTokenVotingPluginRepo,
+            preparedOptimisticSetupData
+        );
+
+        revokeApplyInstallationPermissions(dao);
+
+        // REMOVE THIS CONTRACT AS OWNER
+        revokeOwnerPermission(deployment.dao);
+
+        // DEPLOY OTHER CONTRACTS
+        deployment.publicKeyRegistry = deployPublicKeyRegistry();
+        deployment.delegationWall = deployDelegationWall();
```

Comment:
I don't think we should be deploying the DelegationWall here but in another separate script.
This components is separate from the DAO and should be (most likely) deployed in Taiko rather than mainnet. Otherwise announcing yourself would be super expensive

Reply:
Done